nuxeo-runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/api/Framework.java

```diff
@@ -263,20 +265,10 @@ public static Object lookup(String key) {
      *
      * @param runnable what to run
      * @since 8.4
+     * @since 11.1 this method now takes a {@link ThrowableRunnable}
      */
-    public static void doPrivileged(Runnable runnable) {
-        try {
-            LoginContext loginContext = login();
-            try {
-                runnable.run();
-            } finally {
-                if (loginContext != null) { // may be null in tests
-                    loginContext.logout();
-                }
-            }
-        } catch (LoginException e) {
-            throw new RuntimeException(e);
-        }
+    public static <E extends Throwable> void doPrivileged(ThrowableRunnable<E> runnable) throws E {
```

Comment:
Should we not backport this change and the one in `SQLKeyValueStore` as it changes the byte code?

Reply:
Hm yes indeed. So I'll backport just the needed code.